pandas/core/frame.py

```diff
@@ -4467,16 +4468,34 @@ def query(self, expr: str, *, inplace: bool = False, **kwargs) -> DataFrame | No
            A   B  C C
         0  1  10   10
         """
+        
         inplace = validate_bool_kwarg(inplace, "inplace")
         if not isinstance(expr, str):
             msg = f"expr must be a string to be evaluated, {type(expr)} given"
             raise ValueError(msg)
         kwargs["level"] = kwargs.pop("level", 0) + 1
         kwargs["target"] = None
-        res = self.eval(expr, **kwargs)
 
+        if self.index.duplicated().any():
```

Comment:
This is a similar workaround like the previous try/except solution. Ideally `self.eval(expr, **kwargs)` should be fixed directly, not working around it